src/pybkgmodel/data.py

```diff
@@ -501,23 +537,236 @@ def load_events(cls, file_name, cuts):
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the DL3 file to use.
+
+        Returns
+        -------
+        bool
+            True if fits file according to file extension
+        """
+
+        ext = Path(file_name).suffixes
+        compatible = ".fits" in ext
+
+        return compatible
+
+    @classmethod
+    def get_obs_id(cls, file_name):
+        """Function reads the observation id from the fits file.
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the DL3 file to use.
+
+        Returns
+        -------
+        int
+            Observation ID number of the fits file
+
+        Raises
+        ------
+        RuntimeError
+            In case Observation ID is not found or Events HDU layer is missing. In this case the file is not complient to GADF.
+        """
+
+        obs_id = None
+
+        with fits.open(file_name, memmap=False) as input_file:
+            try:
+                obs_id = int(input_file["EVENTS"].header['OBS_ID'])
+            except Exception as error:
+                raise RuntimeError(f'Can not find observations ID in {file_name}') from error
+
+        return obs_id
+
+    @classmethod
+    def load_events(cls, file_name):
+        """Function to read the events from the DL3 file and to compute the missing variables needed by pybkgmodel
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the DL3 file to use.
+
+        Returns
+        -------
+        dict:
+            A dictionary with the event properties: arrival time, direction, and energy.
+        """
+
+        data_names_mapping = {
+            'EVENT_ID': 'daq_event_number',
+            'RA': 'event_ra',
+            'DEC': 'event_dec',
+            'GAMMANESS': 'gammaness',
+            'ENERGY': 'event_energy',
+        }
+
+        with fits.open(file_name, memmap=False) as input_file, \
+            erfa_astrom.set(ErfaAstromInterpolator(1 * u.s)):
+            try:
+                evt_hdu = input_file["EVENTS"]
+                evt_head = evt_hdu.header
+                evt_data = pandas.DataFrame(evt_hdu.data)
+
+                event_data = {}
+
+                for key, name in data_names_mapping.items():
+
+                    if key in evt_data.keys():
+                        event_data[name] = evt_data[key].to_numpy()
+                        # Fits header keywords start counting from 1
+                        unit_name = f"TUNIT{evt_data.columns.get_loc(key)+1}"
+
+                        try:
+                            event_data[name] *= u.Unit(evt_head[unit_name])
+                        except KeyError:
+                            event_data[name] *= u.one
+
+                # Event times need to be converted from Instrument reference epoch
+                ref_epoch = astropy.time.Time(evt_head['MJDREFI']+evt_head['MJDREFF'], format='mjd')
```

Comment:
Use the two argument form of `astropy.time.Time`, otherwise you will have precision issues:

```
ref_epoch = astropy.time.Time(evt_head['MJDREFI'], evt_head['MJDREFF'], format='mjd')
```

Reply:
also, you need to read the time scale

Reply:
added, though I found that in the header of the DL3 files the TIMESYS is given in all-caps whereas astropy prefers lower cases, is there a reason for why it is all-caps in the LST files?

Reply:
Because that's what the FITS standard is prescribing